packages/language/src/parser/parser-handwritten.ts

```diff
@@ -0,0 +1,840 @@
+import { Parser } from "./parser-types";
+import * as ast from "../syntax-tree/ast";
+import { IRecognitionException, TokenType } from "chevrotain";
+import { finalParserState, ParserState } from "./parser-state";
+import * as tokens from "./tokens";
+import { CstNodeKind } from "../syntax-tree/cst";
+
+export class HandwrittenParser implements Parser<ast.Program, tokens.Token> {
+    private _input: tokens.Token[] = [];
+    set input(value: tokens.Token[]) {
+        this._input = value;
+    }
+    get errors(): IRecognitionException[] {
+        return [];
+    }
```

Comment:
Can we transform this class into a list of individual functions? Since they all pass along the `ParserState` anyway, they have all the context they need available. See also the current iteration of the preprocessor parser.